pkg/jobs/registry.go

```diff
@@ -895,6 +899,141 @@ func (r *Registry) Start(ctx context.Context, stopper *stop.Stopper) error {
 	})
 }
 
+func (r *Registry) createJobsStatsPollerIfNotExists(ctx context.Context) error {
+	if r.knobs.DisableJobsMetricsPolling {
+		return nil
+	}
+
+	for {
+		const TxnName = "Jobs Stats Poller Txn"
+		txn := r.db.NewTxn(ctx, TxnName)
+
+		exists, err := JobExists(ctx, r.ex, txn, func(payload *jobspb.Payload) bool {
+			return payload.Type() == jobspb.TypePollJobsStats
+		})
+		if err != nil {
+			if errors.HasType(err, (*roachpb.TransactionRetryWithProtoRefreshError)(nil)) {
+				continue
+			}
+			return err
+		}
+
+		if exists {
+			return nil
+		}
+
+		metricsJobID := r.MakeJobID()
+		jr := createJobsStatsPollerRecord()
+		if _, err := r.CreateAdoptableJobWithTxn(ctx, jr, metricsJobID, txn); err != nil {
+			if errors.HasType(err, (*roachpb.TransactionRetryWithProtoRefreshError)(nil)) {
+				continue
+			}
+			return err
+		}
+		if err := txn.Commit(ctx); err != nil {
+			if errors.HasType(err, (*roachpb.TransactionRetryWithProtoRefreshError)(nil)) {
+				continue
+			}
+			return err
+		}
+		return nil
+	}
+}
+
+func createJobsStatsPollerRecord() Record {
+	return Record{
+		Description: jobspb.TypePollJobsStats.String(),
+		Details:     jobspb.PollJobsStatsDetails{},
+		Progress:    jobspb.PollJobsStatsProgress{},
+		CreatedBy:   &CreatedByInfo{Name: username.RootUser, ID: username.RootUserID},
+		Username:    username.RootUserName(),
+	}
+}
+
+// PollMetricsTask polls the jobs table for certain metrics at an interval.
+func (r *Registry) PollMetricsTask(ctx context.Context) error {
+
+	var err error
+	updateMetrics := func(ctx2 context.Context, s sqlliveness.Session) {
+		lc, cleanup := makeLoopController(r.settings, pollJobsMetricsInterval, r.knobs.IntervalOverrides.PollMetrics)
+		defer cleanup()
+		for {
+			select {
+			case <-ctx2.Done():
+				err = ctx.Err()
+				return
+			case <-lc.updated:
+				lc.onUpdate()
+			case <-lc.timer.C:
+				lc.timer.Read = true
+				if err = r.updatePausedMetrics(ctx2, s); err != nil {
+					log.Errorf(ctx2, "failed to update paused metrics: %v", err)
```

Comment:
is paused the only type of the job we care about?

Reply:
I think so. What I tried to do in the PR is to create a generic job for polling for metrics, but, at the moment, the only functionality is to poll to count the number of paused jobs. I figured a log message might be useful here if we can't update the metric for some reason.